packages/components/src/DateTimePickers/DateRange/Manager/Manager.component.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { DateRangeContext } from '../Context';
+import {
+	extractRangeParts,
+	extractRangePartsFromTextInput,
+	extractRangePartsFromDate,
+} from '../date-range-extraction';
+
+function ContextualManager(props) {
+	function getOptions() {
+		return {
+			dateFormat: props.dateFormat,
+		};
+	}
+	const initialState = extractRangeParts(props.startDate, props.endDate, getOptions());
+	const [state, setState] = useState(initialState);
+
+	useEffect(() => {
+		if (props.startDate !== state.startDate.value || props.endDate !== state.endDate.value) {
+			const parts = extractRangeParts(props.startDate, props.endDate, getOptions());
+			setState(parts);
+		}
+	}, [props.startDate, props.endDate]);
+
+	function onDatesChange(event, nextState) {
+		if (props.onChange) {
+			const payload = {
+				startDate: nextState.startDate.value,
+				endDate: nextState.endDate.value,
+				errors: nextState.errors,
+				errorMessage: nextState.errorMessage,
+				field: state.focusedInput,
+				origin: 'RANGE_PICKER',
+			};
+			props.onChange(event, payload);
+		}
+	}
+
+	function onFocusChange(event, focusedInput) {
+		setState(prevState => ({
+			...prevState,
+			focusedInput,
+		}));
+	}
+
+	function onSelectDate(event, { date }) {
+		const dateParts = extractRangePartsFromDate(date, state, getOptions());
+		const nextState = {
+			...state,
+			...dateParts,
+		};
+		setState(prevState => ({ ...prevState, ...dateParts }));
+		onDatesChange(event, nextState);
+	}
+
+	function onInputChange(event) {
+		const textInput = event.target.value;
+		const parts = extractRangePartsFromTextInput(textInput, state.focusedInput, getOptions());
+		const nextState = { ...state, ...parts };
+		setState(prevState => ({ ...prevState, ...parts }));
+		onDatesChange(event, nextState);
+	}
+
+	return (
+		<DateRangeContext.Provider
+			value={{
+				startDate: state.startDate,
+				endDate: state.endDate,
+				inputManagement: {
+					onChange: onInputChange,
+					onFocus: onFocusChange,
+					focusedInput: state.focusedInput,
+					placeholder: props.dateFormat,
+				},
+				pickerManagement: {
+					onSubmit: onSelectDate,
+				},
```

Comment:
Can you not base the changes on the focused input please ?
In you manager, separate onStartChange/onEndChange. You can then write explicit code for start and end date, extracting parts and replacing in state just the part that is impacted.
Perhaps you won't need any range parts extraction file. The manager can become just an orchestrator, holding the 2 parts (start and date).

In your component, when using the manager, you get 2 changes callbacks that you pass respectively to start date picker and end date picker.

Reply:
Separate onStart(Input)Change/onEnd(Input)Change callbacks created.
But from your comment, it sounds like you expect the `InputDateRangePicker` will be composed by 2 `InputDatePicker`, is this what you mean? I thought we agreed that `InputDateRangePicker` structure will be like this:
```
<InputDateRangePicker>
    <DateRange.Manager>
        <DateRange.Inputs />
        <DateRange.Picker />
    </DateRange.Manager>
</InputDateRangePicker>
```

Reply:
With this structure, you'll struggle to manage focus. It will be based on props, and you have to `watch the props change` to apply the focus.
If you compose in the same component, the 2 inputs, and the manager, you'll be able to manage the focus based on the `event`.

InputDateRangePicker.component.js
```
<DateRange.Manager>
        <DateRange.Input ref={registerStartRef} />
        <DateRange.Input ref={registerEndRef} />
        <DateRange.Picker />
</DateRange.Manager>
```

That's why I asked you in your inputs PR to merge it with the future InputDateRangePicker

Reply:
ok, understood.
I've removed focusInput from context value.
In my next PR will use 2 input to replace inputs.

Reply:
ok, don't forget to remove the focusedInput from `onFocusChange` in Manager state too in your next PR.